src/citationnet/static/pieChart_vega-lite.js

```diff
@@ -0,0 +1,88 @@
+const forChartDiv = document.getElementById("forChartDiv");
+const forChartDivvis = document.getElementById("forChartDivvis");
+const forChartDivloader = document.getElementById("forChartDivloader");
+
+dragElement(forChartDiv);
+
+function toggleFORChart() {
+    if (forChartDiv.hidden) {
+        showFOR();
+    } else {
+        hideFOR();
+    }
+}
+
+async function showFOR() {
+    document.getElementById("btnFORChart").style.fontWeight = "bolder";
+    forChartDivvis.hidden = true;
+    forChartDivloader.hidden = false;
+    forChartDiv.hidden = false;
+
+    await loadFOR();
+
+    forChartDivvis.hidden = false;
+    forChartDivloader.hidden = true;
+}
+
+function hideFOR() {
+    document.getElementById("btnFORChart").style.fontWeight = "normal";
+    forChartDiv.hidden = true;
+}
+
+async function loadFOR() {
+    const stats = await window.net.getStats();
+    const resp = await fetch("/static/vegaLiteSpec.json");
+    const vegaLiteSpec = await resp.json();
+    const domain = stats.map(value => value.category);
+    const range = stats.map(value => "#".concat(value.color));
+
+    vegaLiteSpec.data.values = stats;
+    vegaLiteSpec.encoding.color.scale.domain = domain;
+    vegaLiteSpec.encoding.color.scale.range = range;
+
+    console.log(vegaLiteSpec);
+
+    vegaEmbed('#forChartDivvis', vegaLiteSpec);
```

Comment:
Consider attaching the click events here in order to avoid polluting the global scope.